core/src/main/scala/io/chrisdavenport/shellfish/FilesOs.scala

```diff
@@ -0,0 +1,281 @@
+
+import scodec.{Codec, codecs}
+import scodec.bits.ByteVector
+
+object FilesOs {
+
+  val files = Files[IO]
+
+  // Read operations:
+
+  /**
+   * Reads the contents of the file at the path and returns it as a String.
+   * @param path
+   *   The path to read from
+   * @return
+   *   The file loaded in memory as a String
+   */
+  def read(path: Path): IO[String] = files.readUtf8(path).compile.string
+
+  /**
+   * Reads the contents of the file at the path and returns it as a ByteVector.
+   * @param path
+   *   The path to read from
+   * @return
+   *   The file loaded in memory as a ByteVector
+   */
+  def readBytes(path: Path): IO[ByteVector] =
+    files.readAll(path).compile.to(ByteVector)
+
+  /**
+   * Reads the contents of the file at the path and returns it line by line as a
+   * Vector of Strings.
+   *
+   * @param path
+   *   The path to read from
+   * @return
+   *   The file loaded in memory as a collection of lines of Strings
+   */
+  def readLines(path: Path): IO[Vector[String]] =
+    files.readUtf8Lines(path).compile.toVector
+
+  /**
+   * Reads the contents of the file and loads it as a type `A` using the
+   * provided codec.
+   * @param A
+   *   The type to read the file as
+   * @param path
+   *   The path to read from
+   * @param codec
+   *   The codec that translates the file contents into the type `A`
+   * @return
+   *   The file loaded in memory as a type `A`
+   * @throws java.lang.Exception
+   *   if the file cannot be read or the contents cannot be parsed into the type
+   *   `A`
+   */
+  def readAs[A](path: Path)(implicit codec: Codec[A]): IO[A] =
+    readBytes(path)
+      .map(bytes => codec.decodeValue(bytes.bits).require)
+      .handleErrorWith { e =>
+        new Exception(s"Failed to read file at $path", e).raiseError[IO, A]
+      }
+
+  /**
+   * The function reads the contents of the file at the path and returns it as a
+   * Stream of Bytes, useful when working with large files.
+   * @param path
+   *   The path to read from
+   * @return
+   *   A Stream of Bytes
+   */
+  def stream(path: Path): Stream[IO, Byte] = files.readAll(path)
+
+  /**
+   * The function reads the contents of the file at the `path` and returns it as
+   * a `Chunk[Bytes]`,
+   * @param path
+   *   The path to read from
+   * @return
+   *   The file loaded in memory as a Chunk of Bytes
+   */
+  def chunk(path: Path): IO[Chunk[Byte]] = files.readAll(path).compile.to(Chunk)
+
+  // Write operations:
+
+  /**
+   * This function overwites the contents of the file at the path with the
+   * contents provided in form of a entire string loaded in memory.
+   *
+   * @param path
+   *   The path to write to
+   * @param contents
+   *   The contents to write to the file
+   * @return
+   *   The number of bytes written
+   */
+  def write(path: Path)(contents: String): IO[Long] =
+    Stream
+      .emit(contents)
+      .covary[IO]
+      .through(files.writeUtf8(path))
+      .compile
+      .count
```

Comment:
This will always return 0. Take a look at signature of `Files[F].writeUtf8`:
```scala
def writeUtf8(path: Path): Pipe[F, String, Nothing] = writeUtf8(path, Flags.Write)
```
The output of that pipe is type `Nothing` and there are no values of that type (i.e. we say it is "uninhabited"). Hence, we're guaranteed there's no output values and hence the count is always 0.

I'd change the return type here to be `IO[Unit]`.

Reply:
Oh my, you are totally right, I didn't noticed that. Thanks on pointing that out! <3